src/Parallel/GlobalCache.hpp

```diff
@@ -461,6 +463,28 @@ GlobalCache<Metavariables>::get_self_proxy() noexcept {
   return this->thisProxy;
 }
 
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wsuggest-attribute=noreturn"
+#endif  // defined(__GNUC__) && !defined(__clang__)
+template <typename Metavariables>
+void GlobalCache<Metavariables>::pup(PUP::er& p) noexcept {
+  p | const_global_cache_;
+  p | parallel_components_;
+  p | mutable_global_cache_proxy_;
+  p | parallel_components_have_been_set_;
+  if (not p.isUnpacking() and mutable_global_cache_ != nullptr) {
+    ERROR(
+        "Cannot serialize the const global cache when the mutable global cache "
```

Comment:
then how do we test the serialization of the const global cache???

Reply:
Currently it only tests when the proxy is set, on the multi-core cache tests.
The alternative (though it feels dangerous) would be to just serialize the pointer, then deserialize the pointer under the anticipation that if you've set the pointer, you're running a single-node test and therefore the pointer should still be valid.
What do you think of doing that and printing a scary warning when it occurs so that production executables are at least somewhat protected from the resulting undefined behavior?